### ISSUE TITLE
Supplying irregular boundary shape for TMR to obtain starting heads

### ISSUE THREAD

**Issue by mnfienen-usgs** [aleaf/modflow-setup#5]
It looks like we are going to have to make a special case for using the outermost boundary to pull starting heads from a parent model. The reason is, if we use `sobel` like suggested by @aleaf in [`here`](https://github.com/aleaf/modflow-setup/blob/59addcbaf6dd33af683e4c2864d67f65ca96700a/mfsetup/lakes.py#L454) we can't rely on `idomain` as being continuous. So, instead, I think we will need to intersect the original `idomain` shapefile with both the parent and the inset models, then map the heads from parent to inset directly. Then I guess use `idomain` as a final check to remove CHDs from inactive or passthrough cells.

Bottom line is, not sure it will be possible to easily generalize a method for lakes and external boundary.

**Comment by aleaf** [comment]
@mnfienen-usgs, do you mean at the edges of the model? If so, couldn't we fix that with `np.pad`?

**Comment by mnfienen** [comment]
ok. the edge-detection from the `get_horizontal_connections` method in `lakes` definitely identifies the edges of `idomain` correctly. Now the challenge is to save out the idomain maximum footprint somehow prior to inactivating the pass-through layers. The idea here is that constant heads (or fluxes for that matter) will be applied at the outer edge of the domain, as indicated by a shapefile for idomain. Any pass through cells get deactivated and this in some layers constant heads will only be applied at the outermost domain edge.

So, I was able to cover assigning the heads once the idomain is available as a maximum footprint, but it turns out the updating of idomain from time to time makes this a challenge. So......can we sort out how to save out the idomain max footprint right after the rasterization step?

**Comment by mnfienen** [comment]
Solved by intersecting the idomain shapefile just before setting up the CHDs.
